src/vivarium/framework/configuration.py

```diff
@@ -9,24 +9,31 @@
 
 """
 from pathlib import Path
+from typing import Union, Dict
 
 import yaml
 
 from vivarium.config_tree import ConfigTree, ConfigurationError
+from vivarium.framework.plugins import DEFAULT_PLUGINS
 
-from .plugins import DEFAULT_PLUGINS
 
+def build_model_specification(model_specification: Union[str, Path, ConfigTree] = None,
```

Comment:
**START HERE**

This is the magic.  Well, some work on the configtree makes the magic work.  But what this does is creates the full model specification from the wide ranges of arguments we might receive on initialization.

Recall that the simulation configuration has four levels (in increasing priority level):
- `base` - lowest level defaults are set here
- `component_configs` - where component configuration defaults are set
- `model_override` - where information from a model specification is set
- `override` - Highest priority, for command line args and interactive setting

So this function does:

1. `_get_default_specification` creates the config tree, fills in the default plugins and any configuration in the user `vivarium.yaml` file into the `base` layer of the configuration.
2. We then set any information in the model specification into the `model_override` layer
3. Additional arguments from the component_configuration, configuration, and plugin_configuration are put in as `override`, allowing someone to easily override a file on disk by handing this function a dictionary of info.